🚀 Scaling GRPO to 70B+ Models and Multi-Node Training with vLLM Server & NCCL Communication (#3094)
huggingface/trl: docs/source/grpo_trainer.md

## Customization

> ## Speed up training with vLLM-powered generation

Reviewer: turbo nit:

```suggestion
### Speed up training with vLLM-powered generation
```

> Generation is often the main bottleneck that makes training slow with online methods. To accelerate generation, you can use [vLLM](https://github.com/vllm-project/vllm), a library that enables fast generation. To enable it, first start a vLLM server using the following command:

Reviewer: ```suggestion
Generation is often the main bottleneck that makes training slow with online methods. To accelerate generation, you can use [vLLM](https://github.com/vllm-project/vllm), a library that enables fast generation. To enable it, first install the package with 

```shell
pip install "vllm==0.7.2"
```

and then start a vLLM server using the following command:
```

```bash
trl vllm-serve --model <model_name>
```

> and pass `use_vllm=True` in the training arguments.

Reviewer: ```suggestion
Then, pass `use_vllm=True` in the training arguments and run the training script:
```
